src/main/java/org/candlepin/subscriptions/metering/jmx/OpenshiftJmxBean.java

```diff
@@ -0,0 +1,174 @@
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.metering.jmx;
+
+import org.candlepin.subscriptions.metering.service.prometheus.PrometheusMetricsPropeties;
+import org.candlepin.subscriptions.metering.service.prometheus.task.PrometheusMetricsTaskManager;
+import org.candlepin.subscriptions.resource.ResourceUtils;
+import org.candlepin.subscriptions.util.ApplicationClock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedOperationParameter;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.util.StringUtils;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Exposes the ability to trigger metering operations from JMX.
+ */
+@ManagedResource
+public class OpenshiftJmxBean {
+
+    private static final Logger log = LoggerFactory.getLogger(OpenshiftJmxBean.class);
+
+    private PrometheusMetricsTaskManager tasks;
+
+    private ApplicationClock clock;
+
+    private PrometheusMetricsPropeties metricPropeties;
+
+    @Autowired
+    public OpenshiftJmxBean(ApplicationClock clock, PrometheusMetricsTaskManager tasks,
+        PrometheusMetricsPropeties metricPropeties) {
+        this.clock = clock;
+        this.tasks = tasks;
+        this.metricPropeties = metricPropeties;
+    }
+
+    @ManagedOperation(description = "Perform openshift metering for a single account.")
+    @ManagedOperationParameter(name = "accountNumber", description = "Red Hat Account Number")
+    public void performOpenshiftMeteringForAccount(String accountNumber) throws IllegalArgumentException {
+        Object principal = ResourceUtils.getPrincipal();
+        log.info("Openshift metering for {} triggered via JMX by {}", accountNumber, principal);
+
+        OffsetDateTime end = clock.now();
+        OffsetDateTime start =
+            getStartDate(end, metricPropeties.getOpenshift().getRangeInMinutes());
+
+        try {
+            tasks.updateOpenshiftMetricsForAccount(accountNumber, start, end);
+        }
+        catch (Exception e) {
+            log.error("Error triggering openshift metering for account {} via JMX.",
+                accountNumber, e);
+        }
+    }
+
+    @ManagedOperation(description = "Perform custom openshift metering for a single account.")
+    @ManagedOperationParameter(name = "accountNumber", description = "Red Hat Account Number")
+    @ManagedOperationParameter(
+        name = "endDate",
+        description = "The end date for metrics gathering (default: now). i.e 2018-03-20T09:12:28Z"
+    )
+    @ManagedOperationParameter(
+        name = "rangeInMinutes",
+        description = "Period of time (before the end date) to start metrics gathering. Must be >= 0."
+    )
+    public void performCustomOpenshiftMeteringForAccount(String accountNumber, String endDate,
```

Comment:
For sanity, let's please validate the dates passed are rounded to whole hours...

Reply:
DONE - Available in next commit.